### ISSUE TITLE
Having issues with hardlinks - mergerfs version: 2.34.1

### ISSUE THREAD

**Issue by voarsh2** [trapexit/mergerfs#1193]
Ref: https://github.com/trapexit/mergerfs/issues/411

I'm having issues with hardlinks with this, and I am not really sure why. I've checked some folders inside /downloads/tvshows and see some hardlinks using -la -al.

I see some hardlinks are here:
![image](https://github.com/trapexit/mergerfs/assets/60437695/5cc640d4-4347-4b03-8aa7-18b1dc5ad172)

But not this example (and likely others):
![image](https://github.com/trapexit/mergerfs/assets/60437695/0f756d58-05a1-4e2e-a378-4037b5054f8a)

My paths:
/mnt/storage/downloads/complete/
/mnt/storage/downloads/tvshows

It seems Sonarr isn't creating hardlinks. And when I try and manually create a hardlink:
```
ln -v /mnt/storage/downloads/complete/The.Mandalorian.S03.WEB-DL.1080p.LF_\[rutor\]/* /mnt/storage/downloads/tvshows/The\ Mandalorian/Season\ 3
```
I get: Invalid cross-device link

My setup:
```
/dev/disk/by-uuid/f1f93305-db71-40f8-9a24-52d9884fb218 /mnt/disk1   ext4 defaults 0 0
/dev/disk/by-uuid/8bb1274d-3832-4fcf-8c08-1eb93bf548f1 /mnt/disk2   ext4 defaults 0 0
/dev/disk/by-uuid/4deb4edc-867c-484a-8af8-7fc850445a37 /mnt/disk3   ext4 defaults 0 0

/mnt/disk1:/mnt/disk2:/mnt/disk3:/mnt/bindmounts/shared2 /mnt/storage fuse.mergerfs defaults,nonempty,allow_other,use_ino,cache.files=off,moveonenospc=true,dropcacheonclose=true,minfreespace=200G,fsname=mergerfs 0 0
```

TL;DR:
3 disks using MergerFS, mounting to /mnt/storage.
Download client and Sonarr mount to /mnt/storage/download(complete/tvshows) - inside the MergerFS mount. Hardlinks don't seem to be working 100%

**Comment by trapexit** [comment]
- https://github.com/trapexit/mergerfs#do-hardlinks-work
- https://github.com/trapexit/mergerfs#rename--link
- https://github.com/trapexit/mergerfs#when-a-program-tries-to-move-or-rename-a-file-it-fails

You haven't set the create policy. Meaning you are using epmfs. Meaning you've configured mergerfs to not pick a branch unless the path exists on it. You don't have both relative paths on the relevant branches so it fails with EXDEV as appropriate. Or you've bind mounted into a container the paths separately which won't work.

**Comment by voarsh2** [comment]
Thanks.
Just to clarify, setting the create policy should be like this?
```
/mnt/disk1:/mnt/disk2:/mnt/disk3:/mnt/bindmounts/shared2 /mnt/storage fuse.mergerfs defaults,nonempty,allow_other,use_ino,cache.files=off,moveonenospc=true,dropcacheonclose=true,minfreespace=200G,fsname=mergerfs,category.create=all 0 0
```

**Comment by trapexit** [comment]
It should just not be a `ep*` policy. I don't know why you'd want to use `all` for create policies but it works. Most people would use rand, pfrd, mfs, etc. Or their `msp` equivalents.

**Comment by voarsh2** [comment]
> It should just not be a `ep*` policy. I don't know why you'd want to use `all` for create policies but it works. Most people would use rand, pfrd, mfs, etc. Or their `msp` equivalents.

Hmm. Need to give those other options a closer look when I have time.

I did download a file and my *arr apps make a hardlink now. So that's a plus.
Running jdupes on all my files to claw back the space the many duplicates ate up because hardlinks weren't being made (full copies).

 Should really dedicate more time, before checking out jdupes I was planning on deleting my 12TB library and starting from scratch after using ```category.create=all``` (when confirming hardlinks were being created).
